ssl/ssl_rsa.c

```diff
@@ -921,11 +921,19 @@ static int ssl_set_cert_and_key(SSL *ssl, SSL_CTX *ctx, X509 *x509, EVP_PKEY *pr
                 goto out;
             } else {
                 /* copy to privatekey from pubkey */
-                EVP_PKEY_copy_parameters(privatekey, pubkey);
+                rv = EVP_PKEY_copy_parameters(privatekey, pubkey);
+                if (rv != 1) {
```

Comment:
You can just use if (!EVP_PKEY_copy_parameters()) - the function should return 1 on success 0 on failure.